includes/Helper/ExportRow.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\WooCommerce\Helper;
+
+use ErrorException;
+use MyParcelNL\Sdk\src\Model\Consignment\AbstractConsignment;
+use WC_Order;
+use WC_Order_Item_Product;
+use WC_Product;
+use WCMP_Export;
+use WCMYPA_Admin;
+use WCMYPA_Settings;
+use WPO\WC\MyParcel\Compatibility\Product as WCX_Product;
+
+class ExportRow
+{
+    public const DEFAULT_PRODUCT_QUANTITY = 1;
+    public const CURRENCY_EURO = 'EUR';
+
+    /**
+     * @var \WC_Order
+     */
+    private $order;
+
+    /**
+     * @var \WC_Product
+     */
+    private $product;
+
+    public function __construct(WC_Order $order, WC_Product $product)
+    {
+        $this->order   = $order;
+        $this->product = $product;
+    }
+
+    /**
+     * @return string
+     * @throws \JsonException
+     */
+    public function getCountryOfOrigin(): string
+    {
+        $defaultCountryOfOrigin   = WCMYPA()->setting_collection->getByName(WCMYPA_Settings::SETTING_COUNTRY_OF_ORIGIN);
+        $productCountryOfOrigin   = WCX_Product::get_meta($this->product, WCMYPA_Admin::META_COUNTRY_OF_ORIGIN, true);
+        $variationCountryOfOrigin = WCX_Product::get_meta(
+            $this->product,
+            WCMYPA_Admin::META_COUNTRY_OF_ORIGIN_VARIATION,
+            true
+        );
+        $fallbackCountryOfOrigin  = WC()->countries->get_base_country() ?? AbstractConsignment::CC_NL;
+
+        return $variationCountryOfOrigin ?: $productCountryOfOrigin ?: $defaultCountryOfOrigin ?: $fallbackCountryOfOrigin;
+    }
+
+    /**
+     * @return int
+     * @throws \ErrorException|\JsonException
+     */
+    public function getHsCode(): int
+    {
+        $defaultHsCode   = WCMYPA()->setting_collection->getByName(WCMYPA_Settings::SETTING_HS_CODE);
+        $productHsCode   = WCX_Product::get_meta($this->product, WCMYPA_Admin::META_HS_CODE, true);
+        $variationHsCode = WCX_Product::get_meta($this->product, WCMYPA_Admin::META_HS_CODE_VARIATION, true);
+
+        $hsCode = $productHsCode ?: $defaultHsCode;
+
+        if ($variationHsCode) {
+            $hsCode = $variationHsCode;
+        }
+
+        if (! $hsCode) {
+            throw new ErrorException(__("No HS code found in MyParcel settings", "woocommerce-myparcel"));
+        }
+
+        return (int) $hsCode;
```

Comment:
Dit is niet voor deze story, maar volgens mij zijn hs codes geen ints. Het zijn strings die bestaan uit cijfers (en spaties voor de leesbaarheid), en kunnen een voorloop 0 hebben.

Reply:
Wij accepteren alleen HS-codes van maximaal 5 character. In deze HS codes zitten dan ook geen spaties